src/decorator/params.ts

```diff
@@ -69,6 +69,32 @@ export function Param(name: string) {
 }
 
 /**
+ * This decorator allows to inject a State object to the controller action parameter.
+ * Applied to class method parameters.
+ *
+ * @param objectName The name of object stored in State
+ */
+export function State(objectName?: string) {
```

Comment:
State is not a session, it's a recomended place for storing data for passing between middlewares:
https://github.com/koajs/koa/blob/master/docs/api/context.md#ctxstate